source/code/providers/support/runasprovider.cpp

```diff
@@ -212,7 +212,7 @@ namespace SCXCore
 
         try
         {
-            SCXFilePath scriptfile = SCXFile::CreateTempFile(script);
+            SCXFilePath scriptfile = CreateTempFile(script);
```

Comment:
I don't care much for creating a small helper to do this. I suggest resolving the tempdir via SystemInfo and then just passing the second parameter to SCXFile::CreateTempFile().

You only get an advantage if the second function is called in multiple places. But it's not, so it's just a needless level of indirection.